test/testenv.py

```diff
@@ -43,6 +37,15 @@ def __init__(self, filename):
         with open(filename, 'r') as f:
             self.options = strictyaml.load(f.read(), options_schema).data
 
+    def get_testid(self):
+        """Convert a testcase filename into a test case identifier."""
+        name = os.path.splitext(os.path.basename(self.filename))[0]
```

Comment:
We're constructing some of these strings repeatedly. We could push some of this stuff to the constructor or something. Not critical performance wise though and it's certainly no worse than before.

Reply:
Pushed a commit fixing this.